### PR TITLE
Handling errors from token endpoint

### LINKED ISSUE #38631: Investigate and handle Jetpack AI JWT token generation failures
When testing the Woo AI plugin, we got a report of a failure to get the Jetpack AI JWT token:
D112749#2250441-code

- I was able to generate name ideas for a product title using AI
- I got an API error when trying to generate a product description /wp-json/jetpack/v4/jetpack-ai-jwt?_cacheBuster=1686107585822&_locale=user.
- There was nothing in the UI that notifies the user of that error, Write with AI button simply froze.

Image

Here's where that error would be returned:
https://github.com/Automattic/jetpack/blob/002366d737be18a20c4b1fc52fddcff681f4bc6a/projects/plugins/jetpack/_inc/lib/class.core-rest-api-endpoints.php#L796

It then hits a WPCOM endpoint for generating + validating the JWT token. It has a `check_permissions()` callback which will make sure the user can edit posts and that Jetpack is authorized for the site.
fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Serfg%2Qncv%2Qcyhtvaf%2Sraqcbvagf%2Swrgcnpx%2Qbcranv%2Qdhrel.cuc%3Se%3Q4n63n939%2376-og

I'm not able to replicate the error at the moment but we could look into more gracefully handling the error.

### PR DESCRIPTION
**Pull request by joelclimbsthings**
### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:
Adding some basic error handling in the scenario that the token request fails, which results in a generic connection error message to the user.

Closes #38631.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. I've been using this with a WoA dev site ([guide](pb5gDS-1rQ-p2)). Following that guide, but with [this modified script](https://gist.github.com/joelclimbsthings/d0144f525e165d4943650feeec09c9a4).
2. Once you've copied the plugin over, ensure the `Woo AI` plugin is activated.
3. Go to Products -> Add new
4. I'm a little uncertain of the best way to test this since it should only occur if an error is thrown. You can just manually throw an error in [this requestToken function](https://github.com/woocommerce/woocommerce/blob/d278e83140915dcffbb7dcb9c39edd4262c67c61/plugins/woo-ai/src/utils/jetpack-completion.ts#L35), like `throw new Error('test error');`
5. If done, observe that when you click "Write with AI" you see an error within the tiny editor.

### CODE REVIEW

File: plugins/woo-ai/src/product-description/product-description-button-container.tsx [hunk 1]
```diff
@@ -18,11 +18,19 @@ import { recordTracksFactory, getPostId } from '../utils';
...
-const getApiError = () => {
-	return __(
-		`❗ We're currently experiencing high demand for our experimental feature. Please check back in shortly.`,
-		'woocommerce'
-	);
+const getApiError = ( error: string ) => {
+	switch ( error ) {
+		case 'connection_error':
+			return __(
+				'❗ We were unable to reach the experimental service. Please check back in shortly.',
```
Comment (joelclimbsthings) [inline plugins/woo-ai/src/product-description/product-description-button-container.tsx:25]: A slightly altered error if the issue was with creating the connection, although probably not actually necessary. Any opinions on what this should be, or if it should just be the same as the default?

Comment (tommyshellberg) [reply plugins/woo-ai/src/product-description/product-description-button-container.tsx:25]: Honestly, I think I like this _new_ error message as the default:
`We were unable to reach the experimental service. Please check back in shortly.`
Unless I'm mistaken, we don't actually have a check for a specific error code that would hint at our AI services being at capacity? If that's the case, I personally like keeping it simple with everything returning "We were unable to reach the experimental service. Please check back in shortly."

**Comment by tommyshellberg** [comment]
Glancing through this, it looks correct to me but I haven't been able to test it yet (I will need to set up a new Woo Express testing site and I'm AFK tomorrow). I can fully test this on Monday if that's okay. 👍

File: plugins/woo-ai/src/product-description/product-description-button-container.tsx (hunk 1 above)
Comment (joelclimbsthings) [reply plugins/woo-ai/src/product-description/product-description-button-container.tsx:25]: True, although we'll presumably have more control over this when we switch to our own endpoints. We could get more granular at that point.

**Review by tommyshellberg — APPROVED** [review]
Nice job here, LGTM! 🚢

**Pull request merged**